gwpy/table/tests/test_table.py

```diff
@@ -505,6 +505,30 @@ def test_hist(self, table):
     def test_get_column(self, table):
         utils.assert_array_equal(table.get_column('snr'), table['snr'])
 
+    def test_cluster(self):
+        # check that the central data points are all clustered away,
+        # the original table is unchanged, and all points return their
+        # intended values
+        t = self.TABLE(data=[[11, 1, 1, 10, 1, 1, 9],
+                       [0.0, 1.9, 1.95, 2.0, 2.05, 2.1, 4.0]],
+                       names=['amplitude', 'time'])
+        t2 = t.cluster(timecolumn='time', clusterparam='amplitude', window=0.6)
+        assert len(t2) == 3
+        assert len(t) == 7
+        assert all(t2.get_column('amplitude') == [11, 10, 9])
+        assert all(t2.get_column('time') == [0.0, 2.0, 4.0])
+
+        # check that a large cluster window returns at least one data point
+        t3 = t.cluster(timecolumn='time', clusterparam='amplitude', window=10)
+        assert len(t3) == 1
+        assert all(t3.get_column('amplitude') == [11])
+        assert all(t3.get_column('time') == [0.0])
+
+        # check that a non-positive window throws an appropriate ValueError
+        with pytest.raises(ValueError) as exc:
+            t.cluster(timecolumn='time', clusterparam='amplitude', window=0)
+        assert str(exc.value) == 'Window must be a positive value'
```

Comment:
Can you split this test into three separate tests, since they are all fairly independent? You might want to set up `clustertable` as a `fixture` so that it can be reused by each of the separate tests.

Reply:
Done, all tests pass from the command line